wallet/src/de/schildbach/wallet/ui/invite/OnboardFromInviteActivity.kt

```diff
@@ -120,8 +131,16 @@ class OnboardFromInviteActivity : AppCompatActivity() {
     }
 
     override fun finish() {
-        setResult(Activity.RESULT_OK)
         super.finish()
         overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
     }
+
+    override fun onBackPressed() {
+        if (mode == Mode.STEP_3) {
+            startActivity(goNextIntent)
+            finish()
+        } else {
+            super.onBackPressed()
+        }
+    }
```

Comment:
Before adding this code, the back button on Step 3 (after the wallet is created) would close the app.  Therefore it would skip over the Secure Wallet step and possibly forget to create the username (not sure about this though).

The simple fix was to have the back button go to the next step in the process.  If the user pressed back on the next screen, they will go to the Home Screen.